central/notifier/processor/singleton.go

```diff
@@ -42,13 +45,37 @@ func initialize() {
 
 	// When alerts are generated, we will want to notify.
 	pr = New(ns, reporter.Singleton())
+
+	cryptoKey := ""
+	if env.EncNotifierCreds.BooleanSetting() {
+		var err error
+		cryptoKey, err = notifierUtils.GetNotifierSecretEncryptionKey()
+		if err != nil {
+			utils.CrashOnError(err)
```

Comment:
Do we want to panic on error? Or should we start panicking after we have added and enabled both ffs? I am thinking if on prem users enable the ff but not put the key at expected location.

Reply:
If the env var is enabled, and the key is not available, a panic would be the right thing to do since notifiers will not work without.